bgpd/bgp_flowspec_util.c

```diff
@@ -581,3 +582,27 @@ int bgp_flowspec_match_rules_fill(uint8_t *nlri_content, int len,
 	}
 	return error;
 }
+
+/* return 1 if FS entry invalid or no NH IP */
+int bgp_flowspec_get_first_nh(struct bgp *bgp, struct bgp_path_info *pi,
+			      struct prefix *p)
+{
+	struct bgp_pbr_entry_main api;
+	int i;
+	struct bgp_node *rn = pi->net;
+	struct bgp_pbr_entry_action *api_action;
+
+	memset(&api, 0, sizeof(struct bgp_pbr_entry_main));
+	if (bgp_pbr_build_and_validate_entry(&rn->p, pi, &api) < 0)
```

Comment:
Doesn't this call modify pi?  If so the function name 'bgp_flowspec_get_first_nh' is misleading as it doesn't imply that  a 'build' can occur.

Reply:
this function does not modify pi.
only api structure is filled in.

Reply:
okay, I misread the pbr_build function